File: src/FitnessTracker.Api/Models/Exercise.cs
using FitnessTracker.Api.Models.Enums;

namespace FitnessTracker.Api.Models;

public class Exercise
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public string Name { get; set; } = string.Empty;

    public ExerciseType ExerciseType { get; set; }

    public ExerciseTrackingType TrackingType { get; set; }

    public string? PrimaryMuscleGroup { get; set; }

    public string? Equipment { get; set; }

    public bool IsCustom { get; set; }

    public Guid? CreatedByUserId { get; set; }

    public ApplicationUser? CreatedByUser { get; set; }

    public bool IsArchived { get; set; }

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

    public ICollection<WorkoutExercise> WorkoutExercises { get; set; }
        = new List<WorkoutExercise>();
}

File: src/fitness-tracker-web/src/types/exercise.ts
export type ExerciseType =
    | 'Strength'
    | 'Cardio';

export type ExerciseTrackingType =
    | 'WeightAndReps'
    | 'RepsOnly'
    | 'Duration'
    | 'DistanceAndDuration';

export interface Exercise {
    id: string;
    name: string;
    exerciseType: ExerciseType;
    trackingType: ExerciseTrackingType;
    primaryMuscleGroup: string | null;
    equipment: string | null;
    isCustom: boolean;
    isArchived: boolean;
    createdAtUtc: string;
}

export interface CreateExerciseRequest {
    name: string;
    exerciseType: ExerciseType;
    trackingType: ExerciseTrackingType;
    primaryMuscleGroup: string | null;
    equipment: string | null;
}

export interface UpdateExerciseRequest {
    name: string;
    exerciseType: ExerciseType;
    trackingType: ExerciseTrackingType;
    primaryMuscleGroup: string | null;
    equipment: string | null;
}